From 4177572ab73cd84a3a0a898bd6457ad05feef820 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Thu, 26 Dec 2024 12:29:36 +0100
Subject: [PATCH 035/126] phy: amlogic: meson8b-usb2: register child (USB
 connector) devices

Populate the child devices/nodes which are typically USB connectors
with a compatible string (and additional properties) like
"gpio-usb-b-connector".

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 drivers/phy/amlogic/phy-meson8b-usb2.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/phy/amlogic/phy-meson8b-usb2.c b/drivers/phy/amlogic/phy-meson8b-usb2.c
index 3f96df0ad..b09778d72 100644
--- a/drivers/phy/amlogic/phy-meson8b-usb2.c
+++ b/drivers/phy/amlogic/phy-meson8b-usb2.c
@@ -12,6 +12,7 @@
 #include <linux/io.h>
 #include <linux/mod_devicetable.h>
 #include <linux/module.h>
+#include <linux/of_platform.h>
 #include <linux/property.h>
 #include <linux/regmap.h>
 #include <linux/reset.h>
@@ -312,6 +313,11 @@ static int phy_meson8b_usb2_probe(struct platform_device *pdev)
 			return ret;
 	}
 
+	ret = devm_of_platform_populate(&pdev->dev);
+	if (ret)
+		return dev_err_probe(&pdev->dev, ret,
+				     "Failed to create child devices/connectors\n");
+
 	phy = devm_phy_create(&pdev->dev, NULL, &phy_meson8b_usb2_ops);
 	if (IS_ERR(phy)) {
 		return dev_err_probe(&pdev->dev, PTR_ERR(phy),
-- 
2.48.1

